skip.js

```diff
@@ -94,13 +96,13 @@ function skipType(buffer, offset, typeid) {
         }));
     }
 
-    return new ReadResult(null, offset);
+    return new ReadResult(null, offset, result && result.value);
 }
 
 function skipStruct(buffer, offset) {
     var result;
+    var ret = {};
```

Comment:
Skip should avoid allocation. It exists to skip and skip alone. The TStruct and T* system exists for constructing structures off the wire in the intermediate representation. It may need a thin wrapper to delegate to the correct RW based on field type.